include/bluetooth/services/otc.h

```diff
@@ -106,21 +123,25 @@ struct bt_otc_instance_t {
 	struct bt_otc_obj_metadata cur_object;
 };
 
-/** @brief Callback used decoding the directory listing.
+/** @brief Directory listing object metadata callback
  *
- *  @param object The object metadata that was decoded.
+ * If a directory listing is decoded using bt_otc_decode_dirlisting(),
+ * this callback will be called for each object in the directory listing.
+ *
+ *  @param meta The metadata of the decoded object
  *
  *  @return int   BT_OTC_STOP or BT_OTC_CONTINUE. BT_OTC_STOP can be used to
  *                stop the decoding.
  */
-typedef int (*bt_otc_dirlisting_cb)(struct bt_otc_obj_metadata *object);
+typedef int (*bt_otc_dirlisting_cb)(struct bt_otc_obj_metadata *meta);
 
+/** OTC callback structure */
 struct bt_otc_cb {
```

Comment:
In `struct bt_ots_cb`, the instance parameter comes first and then `struct bt_conn`, should we keep the same convention here for consistency?

Reply:
Done.
Although in many other cases, the connection is the first parameter.  (In which case I should rather have changed the ots.)
@Thalley What is your opinion?

Reply:
Good question. 
On one hand `conn` is the first argument in all other APIs besides OTS, but on the other hand it certainly makes sense to keep the `ots_client` and `ots` APIs similar. 

I'm tempted to suggest that we do `conn, ots` and update the OTS API while we can. 

Another thing to consider, does the OTS (and client) API need the `conn` point at all? For e.g. most LE Audio services we've actually omitted the `conn` pointer in the APIs and simply have an instance pointer. Is there a usecase for having the `bt_conn` pointer at all?

Reply:
It does not seem to be really needed.
Although if we remove it, we may want to add the connection to the instance instead, as for e.g. vcs.
Anyway, changing the API of OTS seems to be to be outside of the scope of getting the OTC in.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/42732